src/ScrapingInput.re

```diff
@@ -0,0 +1,72 @@
+open Chrome.Extensions;
+
+/** Convert value from event to string
+One of the disadvantages from reason is that you have to do this type casting */
+let valueFromEvent = evt => (
+                              evt
+                              |> ReactEventRe.Form.target
+                              |> ReactDOMRe.domElementToObj
+                            )##value;
+
+type scrapingReturn('a) =
+  | None
+  | Some('a);
+
+/** This is where all the scraping beauty happens
+MAKE SURE THAT WHATEVER SCRIPT YOU'RE PASSING, IT RETURNS A VALUE,
+WHICH YOU CAN THEN MANIPULATE IN PROCESSFN AND VALIDATE IN VALIDATIONFN
+First: validate (return null if error)
+Then: process the data how you want (make sure that the output of this function matches the input of the reducerFn, though, and of your action type)
+Then: update the state of your input field with a reducer function (which is really a dispatcher) */
+let scrape = (script, validationFn, processFn, reducerFn) => {
+  let scriptDetails = Tabs.mkScriptDetails(~code=script, ());
+  Tabs.executeScriptWithCallback(
+    scriptDetails,
+    result => {
+      /** Type casting to option type to get around the bad bindings the original author of the
+      Chrome bindings provided */
+      let result = result[0] |> Obj.magic |> Js.Nullable.toOption;
+      switch (result |> validationFn) {
```

Comment:
You wrote very nice code I was able to read throughout even with my little ReasonML knowledge.
What I am gonna ask may not be feasible, I'm open to discussion:
Can't we write lines 28,29 as:
```
let validatedResult = result[0] |> Obj.magic |> Js.Nullable.toOption |> validationFn;
switch (validatedResult) { 
```

Reply:
don't worry about this one, it's just a matter of taste. I'm ok with it as it is.